subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -3193,36 +3191,54 @@ static inline void event_len_prep(struct ll_conn *conn)
 
 		if (!conn->common.fex_valid ||
 		    (!feature_coded_phy && !feature_phy_2m)) {
-			rx_time = PKT_US(LL_LENGTH_OCTETS_RX_MAX, 0);
+			rx_time = MAX(PKT_US(LL_LENGTH_OCTETS_RX_MAX, 0),
+				      PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0));
 #if defined(CONFIG_BT_CTLR_PHY)
 			tx_time = MIN(PKT_US(LL_LENGTH_OCTETS_RX_MAX, 0),
-				      conn->default_tx_time);
+				      conn->llcp_length.tx_time);
 #else /* !CONFIG_BT_CTLR_PHY */
 			tx_time = PKT_US(conn->default_tx_octets, 0);
 #endif /* !CONFIG_BT_CTLR_PHY */
+
 #if defined(CONFIG_BT_CTLR_PHY)
 #if defined(CONFIG_BT_CTLR_PHY_CODED)
 		} else if (feature_coded_phy) {
-			rx_time = PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(2));
-			tx_time = conn->default_tx_time;
+			rx_time = MAX(PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(2)),
+				      PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0));
+			tx_time = MIN(PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(2)),
+				      conn->llcp_length.tx_time);
+
 #endif /* CONFIG_BT_CTLR_PHY_CODED */
 
 #if defined(CONFIG_BT_CTLR_PHY_2M)
 		} else if (feature_phy_2m) {
-			rx_time = PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(1));
+			rx_time = MAX(PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(0)),
```

Comment:
Shouldn't it be PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(1))  ?  i.e. BIT(1) ? for 2M PHY

Reply:
Let me introduce defines for 2M_PHY, CODED_PHY, 1M_PHY, that will be clearer than BIT(x)

Reply:
I'll fix it here, and add the defines in a separate PR